### ISSUE TITLE
make the plugin threadSafe

### ISSUE THREAD

**Issue by royts** [eirslett/frontend-maven-plugin#132]
hi,
when using maven to build multiple projects, you can ask to run the build in parallel by `-TX` when `X` is the number of threads.

when a plugin is not thread-safe the build will take more time to run.
from maven build output:
```
[WARNING] * Your build is requesting parallel execution, but project      *
[WARNING] * contains the following plugin(s) that are not marked as       *
[WARNING] * @threadSafe to support parallel building.                     *
[WARNING] * While this /may/ work fine, please look for plugin updates    *
[WARNING] * and/or request plugins be made thread-safe.                   *
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against maven-core                              *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked @threadSafe in MyProject:
[WARNING] com.github.eirslett:frontend-maven-plugin:0.0.20
[WARNING] *****************************************************************
```

is `frontend-maven-plugin` thread-safe?
seems like [marking a plugin as thread-safe is easy](https://maven.apache.org/plugin-tools/maven-plugin-annotations/apidocs/org/apache/maven/plugins/annotations/Mojo.html#threadSafe()).
can we do that?

**Comment by eirslett** [comment]
I'm not sure whether it's thread-safe or not... I guess that would be up to the build of the individual user. Some grunt builds could run in parallell, others would interfere with eachother. What's the common practice for mojos when you're not sure?

**Comment by eirslett** [comment]
I'll close this due to inactivity.

**Comment by Stephan202** [comment]
@eirslett, for a long time Maven stopped displaying this warning, but with [MNG-6171](https://issues.apache.org/jira/browse/MNG-6171) fixed and Maven 3.5.0 released last Friday, users building with `-T` will start seeing this warnings again.

In today's word of multi-core machines it's common to build a multi-module Maven build with `-T`. Have you since figured our whether the plugin is officially safe for concurrent execution? (Unofficially I can say it seems to always just have worked.)

**Comment by Stephan202** [comment]
Doh, should have read #466 first. Well, if concurrent execution is now officially supported, then all that remains is to annotate the plugin accordingly :)

**Comment by mriehema** [comment]
As you can see, I made` install-node-and-npm` threadsafe. Running npm, grunt, gulp or any other execution, they probably not threadsafe yet.

**Comment by Stephan202** [comment]
@mriehema, gotcha. We're using `npm` in addition to `install-node-and-npm`. Should/shall I file a separate ticket to ask for that goal to be made thread-safe?

**Comment by mriehema** [comment]
Yes please and feel free to open a pull request with a fix.

**Comment by mtraynham** [comment]
I've been getting this warning since 1.5 as well. 

Personally, I think we should mark **all plugins as threadSafe, even the Yarn goal**.  What Maven means by "thread safe" here implies that the plugin goal can be run concurrently on multiple modules (i.e. directories).  This is not necessarily a problem with single modules as all goals still run synchronously within that module.

It's usually ok to have two submodule Gulp/Grunt processes running at the same time, as long as they are acting on different modules and not influencing each other.  On the contrary, it's **not** ok to have multiple Yarn processes running at the same time as the Yarn cache could be altered for one project and have effect on another.  There is a Yarn workaround, which is to use a mutex file as suggested [here](https://yarnpkg.com/lang/en/docs/cli/#toc-concurrency-and-mutex).

Because it's up to the user to run concurrently with the `-T` flag, and there is a Yarn workaround, I would suggest just marking all goals as `threadSafe`.
